packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -102,10 +136,163 @@ class CupertinoContextMenu extends StatefulWidget {
   CupertinoContextMenu({
     super.key,
     required this.actions,
-    required this.child,
-    this.previewBuilder,
+    this.child,
+    this.builder,
+    @Deprecated(
+      'Use CupertinoContextMenu.builder instead. '
+      'This feature was deprecated after v3.4.0-34.1.pre.',
+    )
+    this.previewBuilder = _defaultPreviewBuilder,
   }) : assert(actions != null && actions.isNotEmpty),
-       assert(child != null);
+       assert(child != null || builder != null,
+        'Either a child or a builder is required'
+       ),
+       _builder = builder ?? ((BuildContext context, Animation<double> animation) => child!);
+
+  /// Exposes the default border radius for matching iOS 16.0 behavior. This
+  /// value was eyeballed from the iOS simulator running iOS 16.0.
+  ///
+  /// {@tool snippet}
+  ///
+  /// Below is example code in order to match the default border radius for an
+  /// iOS 16.0 open preview.
+  ///
+  /// ```dart
+  /// CupertinoContextMenu(
+  ///   actions: <Widget>[
+  ///     CupertinoContextMenuAction(
+  ///       child: const Text('Action one'),
+  ///       onPressed: () {},
+  ///     ),
+  ///   ],
+  ///   builder: (BuildContext context, Animation<double> animation) {
+  ///     return FittedBox(
+  ///       fit: BoxFit.cover,
+  ///       child: ClipRRect(
+  ///         borderRadius: BorderRadius.circular(CupertinoContextMenu.kOpenBorderRadius * animation.value),
+  ///         child: Image.asset('assets/photo.jpg'),
+  ///       ),
+  ///     );
+  ///   },
+  /// )
+  /// ```
+  ///
+  /// {@end-tool}
+  static const double kOpenBorderRadius = _previewBorderRadiusRatio;
+
+  /// Exposes the final box shadow of the opening animation of the child widget
+  /// to match the default behavior of the native iOS widget. This value was
+  /// eyeballed from the iOS simulator running iOS 16.0.
+  static const List<BoxShadow> kEndBoxShadow = _endBoxShadow;
+
+
+  /// The point at which the CupertinoContextMenu begins to animate
+  /// into the open position.
+  ///
+  /// A value between 0.0 and 1.0 corresponding to a point in [builder]'s
+  /// animation.
```

Comment:
Nit: Maybe explain how someone might use this (or point to another place in the docs where it's explained).